Automattic/pocket-casts-ios
# Feature/updade account upgrade flows

Fixes #

This PR updates the upgrade prompt on Account to use the same cards that we have on the onboarding flow.
This allows us to go directly to Apple IAP flow from the account screen.

In order to test what option result in better conversion this feature will be controlled trough a feature flag: `new_account_upgrade_prompt_flow`

| 1 | 2 | 3 | 4 |
| - | - | - | - |
| ![Simulator Screenshot - iPhone 15 Pro - 2024-02-06 at 17 05 13](https://github.com/Automattic/pocket-casts-ios/assets/651601/60442725-eabd-4302-bdca-2c8dcb283e71) |  ![Simulator Screenshot - iPhone 15 Pro - 2024-02-06 at 17 05 19](https://github.com/Automattic/pocket-casts-ios/assets/651601/32da364c-2e73-46b6-887a-c994b38e1caa) |  ![Simulator Screenshot - iPhone 15 Pro - 2024-02-06 at 17 05 23](https://github.com/Automattic/pocket-casts-ios/assets/651601/34f05d5b-b027-4339-a65e-a0bfcd5e7670) |  ![Simulator Screenshot - iPhone 15 Pro - 2024-02-06 at 17 05 28](https://github.com/Automattic/pocket-casts-ios/assets/651601/a0b07c34-00e4-4971-950c-60c43f1a674d) | 


## To test

 - Edit the podcast scheme to use the "Pocket Casts Configuratio.storekit" file for the StoreKitConfiguration
 - Start the app on the simulator
 - Enable the FF `new_account_upgrade_prompt_flow` in the developer settings
 - Go to the Profile tap ( you must have an account with no active subscription)
 - Tap on Account
 - Check if the cards show correctly for the upgrade options like the screenshots above
 - Switch between the Yearly and Monthly options and see if it shows correctly
 - Tap on Subscribe to Plus
 - Check if the IAP flow starts correctly
 - You can repeat the tests above for Patron 

## Checklist
